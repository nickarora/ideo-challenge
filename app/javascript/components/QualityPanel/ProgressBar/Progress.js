import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'

export const Bar = styled.div`
  height: 100%;
`

const Progress = ({ progress, color }) => (
  <Bar
    style={{
      background: color,
      width: `${Math.abs(progress)}%`,
    }}
  />
)

Progress.propTypes = {
  progress: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
}

export default Progress
