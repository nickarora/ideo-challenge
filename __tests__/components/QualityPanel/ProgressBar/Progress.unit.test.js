import Progress from '~/components/QualityPanel/ProgressBar/Progress'

describe('Progress', () => {
  it('renders with a given progress and color', () => {
    const wrapper = shallow(<Progress progress={50} color={'#333'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
