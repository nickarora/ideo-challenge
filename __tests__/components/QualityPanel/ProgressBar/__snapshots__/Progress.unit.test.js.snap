// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders with a given progress and color 1`] = `
<styled.div
  style={
    Object {
      "background": "#333",
      "width": "50%",
    }
  }
/>
`;
